packages/compiler/core/checker.ts

```diff
@@ -1756,10 +1894,43 @@ export function createChecker(program: Program): Checker {
   }
 
   function cloneType<T extends Type>(type: T, additionalProps: { [P in keyof T]?: T[P] } = {}): T {
-    return createType({
-      ...type,
-      ...additionalProps,
-    });
+    // TODO: this needs to handle other types
```

Comment:
It works for the projectable things but probably other types could use special treatment here.